frontend/app/src/types/defi/compound.ts

```diff
@@ -38,39 +45,45 @@ export const COMPOUND_EVENT_TYPES = [
   DEFI_EVENT_LIQUIDATION,
   'comp'
 ] as const;
-export type CompoundEventType = typeof COMPOUND_EVENT_TYPES[number];
-
-interface CompoundEvent {
-  readonly eventType: CompoundEventType;
-  readonly address: string;
-  readonly blockNumber: number;
-  readonly timestamp: number;
-  readonly asset: string;
-  readonly toAsset?: string;
-  readonly value: Balance;
-  readonly toValue?: Balance;
-  readonly realizedPnl?: Balance;
-  readonly txHash: string;
-  readonly logIndex: number;
-}
 
-type CompoundAssetProfitAndLoss = Readonly<Record<string, Balance>>;
+const CompoundEventType = z.enum(COMPOUND_EVENT_TYPES);
+export type CompoundEventType = z.infer<typeof CompoundEventType>;
 
-export type CompoundProfitAndLoss = Readonly<
-  Record<string, CompoundAssetProfitAndLoss>
->;
+const CompoundEvent = z.object({
+  eventType: CompoundEventType,
+  address: z.string(),
+  blockNumber: z.number(),
+  timestamp: z.number(),
+  asset: z.string(),
+  toAsset: z.string().nullable(),
```

Comment:
`nullable()` is `| null`.
But the previous scheme is `| undefined` which should be`optional()` (?)

Reply:
It was wrong though, I tried optional but the response was returning null instead.